pkg/querier/queryrange/roundtrip_test.go

```diff
@@ -202,6 +202,42 @@ func TestLogFilterTripperware(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestInstantQueryTripperware(t *testing.T) {
+
+	tpw, stopper, err := NewTripperware(testConfig, util_log.Logger, fakeLimits{}, chunk.SchemaConfig{}, 0, nil)
+	if stopper != nil {
+		defer stopper.Stop()
+	}
+	require.NoError(t, err)
+	rt, err := newfakeRoundTripper()
+	require.NoError(t, err)
+	defer rt.Close()
+
+	lreq := &LokiInstantRequest{
+		Query:     `sum by (job) (bytes_rate({cluster="dev-us-central-0"}[15m]))`,
+		Limit:     1000,
+		Direction: logproto.FORWARD,
+		Path:      "/loki/api/v1/query",
+	}
+
+	ctx := user.InjectOrgID(context.Background(), "1")
+	req, err := LokiCodec.EncodeRequest(ctx, lreq)
+	require.NoError(t, err)
+
+	req = req.WithContext(ctx)
+	err = user.InjectOrgIDIntoHTTPRequest(ctx, req)
+	require.NoError(t, err)
+
+	count, h := promqlResult(streams)
+	rt.setHandler(h)
+	resp, err := tpw(rt).RoundTrip(req)
+	require.Equal(t, 1, *count)
+	require.NoError(t, err)
+
+	lokiResponse, err := LokiCodec.DecodeResponse(ctx, resp, lreq)
+	require.IsType(t, &LokiResponse{}, lokiResponse) // Is LokiPromResponse in Docker compose
```

Comment:
This is a little weird. `LokiCodec.DecodeResponse` is returning a `LokiResponse`. However, when I run the query through Grafana it expects and gets a `LokiPromReponse` :man_shrugging:

Reply:
Yeah, this will probably need some digging :/ . It should be `LokiPromResponse` as that's what we're apparently encoding it as.

Reply:
I set `count, h := promqlResult(streams)` instead of `count, h := promqlResult(vector)`.